components/file_system/src/file.rs

```diff
@@ -159,7 +159,95 @@ impl File {
     }
 
     pub fn allocate(&self, len: u64) -> io::Result<()> {
-        self.inner.allocate(len)
+        // Bypass fs2::FileExt::allocate because of its incorrect error handling.
```

Comment:
wait the upstream to merge it?

Reply:
upstream has no activity for 4 years now, I'm feeling negative about it. Plus we can always revert the change if upstream merges it afterwards.

Reply:
maybe use our fork?

Reply:
I think that would be an overkill, we don’t have other changes in the future.

Reply:
use your branch is enough, no need to push to tikv repo

Reply:
Oh, make sense.